scripts/register_qemu_binfmt.bats

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bats
+
+load assertions
+
+setup() {
+  SCRIPT_ROOT="$PWD"
+}
+
+skip_unless_ci() {
+  if [ "$CI" != true ]; then 
+    skip
+  fi
+}
+
+@test "qemu is registered in binfmt_misc" {
+  # register_qemu_binfmt will mutate the underlying host system, which we don't want
+  #  to do without fair warning. In addition, on MacOS, this will either have no effect
+  #  on the DockerVM, as it's already configured that way, or it could possibly harm 
+  #  the machine, if it was built with a different version of buildkit/qemu than what 
+  #  this installs. If the setup has no effect, then the tests aren't actually useful. 
+  #  In addition, testing the state of the DockerVM would require additional logic and 
+  #  setup causing the local instances of the tests to differ from what would be run in 
+  #  CI.  If it is desireable to run this particular test locally, it is recommended to
+  #  obtain a VM (ex: with Vagrant) and run this test suite with `CI=true` set in the 
+  #  environment
+  skip_unless_ci
```

Comment:
I am 100% behind gating this test due to its destructive potential. And thanks for the comment!

In fact, maybe even in CI we don't want this necessarily running alongside other tests, as it seems it could affect them? (Especially if running them in parallel which is a nice option to keep hold of for later optimisation.) Could we gate with an explicit opt-in flag maybe e.g. `RUN_DESTRUCTIVE_TESTS=true`?